polkadot/runtime/common/src/slots/mod.rs

```diff
@@ -142,6 +158,11 @@ pub mod pallet {
 		ParaNotOnboarding,
 		/// There was an error with the lease.
 		LeaseError,
+		/// The signing account has no permission to do the operation.
+		NoPermission,
+		/// The signing account has privileges for this operation but the preconditions are not
+		/// met.
+		NotAllowed,
```

Comment:
maybe more descriptive: 
```suggestion
		PreconditionNotMet,
```

Reply:
Yeah this is better 👍